packages/fdc3-agent-proxy/src/channels/DefaultChannel.ts

```diff
@@ -24,47 +23,57 @@ export class DefaultChannel implements Channel {
   }
 
   async broadcast(context: Context): Promise<void> {
-    const request: BroadcastRequest = {
-      meta: this.messaging.createMeta(),
-      payload: {
-        channelId: this.id,
-        context,
-      },
-      type: 'broadcastRequest',
-    };
-    await this.messaging.exchange<BroadcastResponse>(request, 'broadcastResponse');
+    const done = await this.messaging.exchange<BroadcastResponse>(
+      {
+        meta: this.messaging.createMeta(),
+        payload: {
+          channelId: this.id,
+          context,
+        },
+        type: 'broadcastRequest',
+      } as BroadcastRequest,
+      'broadcastResponse'
+    );
+    console.log('broadcast done', done);
   }
 
   async getCurrentContext(contextType?: string | undefined): Promise<Context | null> {
-    const request: GetCurrentContextRequest = {
-      meta: this.messaging.createMeta(),
-      payload: {
-        channelId: this.id,
-        contextType: contextType ?? null,
-      },
-      type: 'getCurrentContextRequest',
-    };
-
     // first, ensure channel state is up-to-date
-    const response = await this.messaging.exchange<GetCurrentContextResponse>(request, 'getCurrentContextResponse');
+    const response = await this.messaging.exchange<GetCurrentContextResponse>(
+      {
+        meta: this.messaging.createMeta(),
+        payload: {
+          channelId: this.id,
+          contextType: contextType ?? null,
+        },
+        type: 'getCurrentContextRequest',
+      } as GetCurrentContextRequest,
+      'getCurrentContextResponse'
+    );
 
     return response.payload.context ?? null;
   }
 
-  async addContextListener(contextType: any, handler?: ContextHandler): Promise<Listener> {
+  async addContextListener(
+    contextTypeOrHandler: string | null | ContextHandler,
+    handler?: ContextHandler
+  ): Promise<Listener> {
     let theContextType: string | null;
     let theHandler: ContextHandler;
 
-    if (contextType == null) {
+    if (contextTypeOrHandler == null && handler) {
       theContextType = null;
-      theHandler = handler as ContextHandler;
-    } else if (typeof contextType === 'string') {
-      theContextType = contextType;
-      theHandler = handler as ContextHandler;
-    } else {
+      theHandler = handler;
+    } else if (typeof contextTypeOrHandler === 'string' && handler) {
+      theContextType = contextTypeOrHandler;
+      theHandler = handler;
+    } else if (contextTypeOrHandler) {
```

Comment:
So, if someone calls `.addContextListener("fdc3.instrument")`, the context type will get null'd out because I didn't provide a handler? Am I reading that right?

Reply:
This is handling the old deprecated addContextListener signature here: https://github.com/finos/FDC3/blob/d5aab3f3cc1d98598cae9d6cdd063b36e509bcdf/src/api/DesktopAgent.ts#L554-L558

It still exists until fdc3 3.0 when it will be terminated (with prejudice). Untilt hen if they only provide one argument then we have to assume its the handler function and that it should be applied to all context types (as if the first arg was null and the second was the handler).

What could make this better is checking that the first argument is a function (which I'll do) - as this call makes no sense without a handler function being passed:
`} else if (typeof contextTypeOrHandler === "function") {`

Hence if they call `.addContextListener("fdc3.instrument")` it'll reject with the error message.